File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/section.model.ts
export interface ISectionModel {
  id?: string;
  description: string;
  courseId: string;
  title: string;
  order: number;
  classRunId: string | null;
  creditsAward?: number;
}

export class SectionModel implements ISectionModel {
  public id?: string;
  public description: string = '';
  public courseId: string = '';
  public title: string = '';
  public order: number = 0;
  public classRunId: string | null = null;
  public creditsAward?: number = 0;

  constructor(data?: ISectionModel) {
    if (data) {
      this.id = data.id;
      this.description = data.description;
      this.courseId = data.courseId;
      this.title = data.title;
      this.order = data.order;
      this.classRunId = data.classRunId;
      this.creditsAward = data.creditsAward;
    }
  }
}

File: pdlearning/src/microservices/course-microservice/Api/Application/Models/SectionModel.cs
using System;
using Microservice.Course.Domain.Entities;

namespace Microservice.Course.Application.Models
{
    public class SectionModel
    {
        public Guid Id { get; set; }

        public string Description { get; set; }

        public Guid CourseId { get; set; }

        public Guid? ClassRunId { get; set; }

        public string Title { get; set; }

        public int Order { get; set; }

        public int? CreditsAward { get; set; }

        public static SectionModel Create(Section section)
        {
            return new SectionModel
            {
                Id = section.Id,
                Description = section.Description,
                Title = section.Title,
                CourseId = section.CourseId,
                ClassRunId = section.ClassRunId,
                Order = section.Order.GetValueOrDefault(),
                CreditsAward = section.CreditsAward
            };
        }
    }
}
